Products/ZenCollector/daemon.py

```diff
@@ -832,53 +832,51 @@ def _startMaintenance(self, ignored=None):
         maintenanceCycle = MaintenanceCycle(interval, heartbeatSender, self._maintenanceCycle)
         maintenanceCycle.start()
 
+    @defer.inlineCallbacks
     def _maintenanceCycle(self, ignored=None):
         """
         Perform daemon maintenance processing on a periodic schedule. Initially
-        called after the daemon configuration loader task is added, but afterward
-        will self-schedule each run.
+        called after the daemon configuration loader task is added,
+        but afterward will self-schedule each run.
         """
         self.log.debug("Performing periodic maintenance")
+        ret = None
+        try:
+            if not self.options.cycle:
+                ret = "No maintenance required"
+            elif getattr(self.preferences, 'pauseUnreachableDevices', True):
+                # TODO: handle different types of device issues, such as WMI issues
+                issues = yield self.getDevicePingIssues()
+                ret = yield self._processDeviceIssues(issues)
+        except Exception as err:
+            self.log.exception('failure in _maintenanceCycle')
+            raise err
```

Comment:
Just need `raise` here, I think.  `raise err` deletes the traceback.

Reply:
Fixed